### ISSUE TITLE
New Rule: valid-css-selector

### ISSUE THREAD

**Issue by Mikilll94** [eslint/eslint#17124]
### Rule details

Check if a css selector is valid

### Related ECMAScript feature

https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector

### What type of rule is this?

Warns about a potential problem

### Example code

```js
document.querySelector('[type="submit"');
document.querySelector('#32131');
```

### Why should this rule be in the core instead of a plugin?

This rule should check if the argument passed to `document.querySelector`, `document.querySelectorAll` is a valid CSS selector.

I have not found any plugins which implement this check.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by aladdin-add** [comment]
Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.
